packages/blocks/src/store/reducer.js

```diff
@@ -377,13 +377,22 @@ export function blockBindingsSources( state = {}, action ) {
 			return {
 				...state,
 				[ action.name ]: {
-					label: action.label,
+					// Don't override the label if it's already set.
+					label: state[ action.name ]?.label || action.label,
 					getValues: action.getValues,
 					setValues: action.setValues,
 					getPlaceholder: action.getPlaceholder,
 					canUserEditValue: action.canUserEditValue,
 				},
 			};
+		case 'ADD_BOOTSTRAPPED_BLOCK_BINDINGS_SOURCE':
+			return {
+				...state,
+				[ action.name ]: {
```

Comment:
See my https://github.com/WordPress/gutenberg/pull/63470#issuecomment-2242387646. I predict it’s the missing edge case to cover. In the case bootstrapping happens after registration, it would replace the entry with data from the server. A unit test would also help here.

Reply:
That makes sense. I'll work on a follow-up pull request to cover that use case.

Reply:
I have started this pull request trying to address this use case: [link](https://github.com/WordPress/gutenberg/pull/63797).